### PR TITLE
Alternative timeline display

### PR DESCRIPTION
**Pull request by hstandaert**
## Description
Implementation of the alternative timeline display. Included changes can be found below..

(This PR includes both the changes for styleguide and drupal implementation)
### 1. New organism `Programme`
* **Styleguide**: A new organism `programme` has been added
* **Drupal 8**: The `timeline frame` content type has been used 

### 2. refactored `image-gallery` and `timeline` components. 
Timeline component has some styling overrides for the image gallery. These overrides were also applied in the design of the (new) programme component, so they were removed from the `timeline` organism and added to the `image gallery` as a variant

### 3. New `Modal` component
A new component called `modal` has been added with 2 variants: 
* Default modal with dynamic height
* Fixed height version where the close-button and submit buttons are displayed fixed on the screen. Only the content in between is scrollable (when applicable)

### 4. Option to change the URL fragment when opening a Modal
The javascript code for the Modal has been extended. You can now pass in the `changeHash` option to change the behavior of the modal. 

* When opening a modal, the fragment of the URL is updated based on the `id` of the HTML element that has the `modal` class. A new state is pushed to the history so the user can press the browser's back-button to close the modal (as well as click the close button in the modal itself).
* When closing the modal, the fragment is removed from the URL
* When opening a page with a fragment, the desired Modal is detected and opened. A new state is pushed to the history so the user can press the browser's back-button to close the modal.

### 5. Bugfixes
* Fix Fatal error in contact block (see change in `field--field-contacts.html.twig`)
* Fix notice "Only variables should be passed in by reference" (see `theme.inc#L73-76` )

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have updated the documentation accordingly.
- [x] I have updated the style guide CHANGELOG accordingly.
- [x] I have updated the gent_base CHANGELOG accordingly.
- [x] I have ran gulp axe on the compiled files and found no errors.

### CODE REVIEW

File: gent_base.theme [hunk 1]
```diff
@@ -323,25 +323,39 @@ function gent_base_preprocess_field(&$variables) {
...
     $isTimeline = FALSE;
+    $isProgram = FALSE;
 
     foreach ($paragraphs as $key => $paragraph) {
       $entity = Paragraph::load($paragraph->get('target_id')->getValue());
       if ($entity) {
         $type = $entity->get('type')->getString();
 
         if ($type === 'timeline_frame') {
+          $isProgram = _gent_base_get_timeline_display($entity);
```
Comment (delrueba) [inline gent_base.theme]: Can you turn this into a boolean and simplify line 340?
$isProgram = _gent_base.... === 'alt'

File: gent_base.theme [hunk 2]
```diff
@@ -599,6 +613,23 @@ function gent_base_preprocess_paragraph__image(&$variables) {
   }
 }
 
+/**
+ * Implements hook_preprocess_HOOK().
+ *
+ * Isolate one of the selected images for the teaser (before opening the
+ * popup). Currently, this'll be the first image of the gallery.
```
Comment (delrueba) [inline gent_base.theme]: Thanks for the documentation! :+1: 
I'd use 'modal' instead of 'popup', since that's also the term we use for the library etc.

File: styleguide/components/11-base/base/modal.functions.js [hunk 3]
```diff
@@ -76,12 +77,34 @@
         options.resizeEvent();
         addResizeEvent();
       }
+
+      /*
```
Comment (delrueba) [inline styleguide/components/11-base/base/modal.functions.js]: Move to new function. 
Too much logic in init.

File: styleguide/components/41-organisms/image-collections/image-gallery/image-gallery.config.js [hunk 4]
```diff
@@ -19,155 +56,87 @@ module.exports = {
...
-            src: 'https://via.placeholder.com/800x500&text=8:5+(800x500)',
-            alt: 'placeholder image alternative text',
-            caption: 'image caption'
-          },
-          {
-            src: 'https://via.placeholder.com/800x500&text=8:5+(800x500)',
-            alt: 'placeholder image alternative text'
-          }
-        ]
+        items: imageList.slice(-2)
```
Comment (delrueba) [inline styleguide/components/41-organisms/image-collections/image-gallery/image-gallery.config.js:38]: Nice refactor! :1st_place_medal:

File: scripts/yarn.lock [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (meirege) [inline scripts/yarn.lock]: Is this file necessary? :)

File: styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss [hunk 6]
```diff
@@ -150,4 +150,181 @@
...
+      margin: 0 -1vw;
+
+      // overrule single image styling
+      .gallery-link {
+        width: 100%;
+        margin: 0;
+      }
+
+      li {
+        width: calc((100% / 1) - 2vw);
```
Comment (meirege) [inline styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss]: Why divide 100% with 1? :P

File: styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss [hunk 7]
```diff
@@ -150,4 +150,181 @@
...
+
+        // hide show more if there is only one image
+        &:first-of-type:nth-last-of-type(-n+2) ~ .image-gallery__show-more {
+          display: none;
+        }
+      }
+
+      // Media Queries
+      @include mobile {
+        max-width: 270px;
```
Comment (meirege) [inline styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss]: Rems?

File: styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss [hunk 8]
```diff
@@ -150,4 +150,181 @@
...
+      }
+
+      @include phablet {
+        max-width: calc(270px * 2 + 2vw);
```
Comment (meirege) [inline styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss]: Rems?

File: styleguide/components/41-organisms/image-collections/image-gallery/image-gallery--multiple.twig [hunk 9]
```diff
@@ -1,4 +1,4 @@
-<div class="image-gallery--wrapper multiple">
+<div class="image-gallery--wrapper multiple{% if type %} image-gallery--{{ type }}{% endif %}">
```
Comment (meirege) [inline styleguide/components/41-organisms/image-collections/image-gallery/image-gallery--multiple.twig:1]: I'd put the space before the {% if... %} and not inside it.

File: styleguide/components/41-organisms/image-collections/image-gallery/image-gallery.config.js [hunk 10]
```diff
@@ -1,5 +1,42 @@
 'use strict';
 
+const imageList = [
```
Comment (meirege) [inline styleguide/components/41-organisms/image-collections/image-gallery/image-gallery.config.js]: Can't we make this dynamic?

File: styleguide/components/41-organisms/image-collections/image-gallery/image-gallery.twig [hunk 11]
```diff
@@ -1,4 +1,4 @@
-<div class="image-gallery--wrapper single">
+<div class="image-gallery--wrapper single{% if type %} image-gallery--{{ type }}{% endif %}">
```
Comment (meirege) [inline styleguide/components/41-organisms/image-collections/image-gallery/image-gallery.twig:1]: Same comment as above with the space positioning.

File: styleguide/components/41-organisms/programme/_programme.scss [hunk 12]
```diff
@@ -0,0 +1,194 @@
+.programme {
+  width: calc(100% + 2 * #{$gutter-width});
+  margin-right: -$gutter-width;
```
Comment (meirege) [inline styleguide/components/41-organisms/programme/_programme.scss:3]: @delrueba Don't we have a mixin for this behaviour?

File: styleguide/components/41-organisms/programme/programme.twig [hunk 13]
```diff
@@ -0,0 +1,95 @@
+<div class="programme-wrapper">
+  <h2 class="visually-hidden">Programme</h2>
+  <dl class="programme box-top">
+    <div class="inner-box">
+
+      {% for slotKey, slot in slots %}
+        <div class="programme-slot" role="listitem">
```
Comment (delrueba) [inline styleguide/components/41-organisms/programme/programme.twig]: I'm afraid you can't have nested divs inside a description list.
https://checker.html5.org/ 
https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element

File: styleguide/components/41-organisms/programme/_programme.scss [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+      font-size: 1.3rem;
+      text-align: center;
+    }
+
+    > dd {
+      max-width: none;
+      padding: 0;
+    }
+
+    & + & {
```
Comment (meirege) [inline styleguide/components/41-organisms/programme/_programme.scss:38]: This is a bid obscure don't you think? It's okay for me but maybe add a comment for people who are not super familiar with SASS? :)

File: styleguide/components/41-organisms/programme/_programme.scss [hunk 15]
```diff
@@ -0,0 +1,194 @@
...
+      margin-top: 2.8rem;
+    }
+  }
+
+  ///
+  /// Programme teaser.
+  ///
+  &-teaser {
+    @include theme('background-color', 'color-secondary');
```
Comment (meirege) [inline styleguide/components/41-organisms/programme/_programme.scss]: Just nitpicking here but I usually add a newline after includes for readability.

File: styleguide/components/41-organisms/programme/programme.twig [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+              {% set key = slotKey ~ itemKey %}
+              <li>
+                <!-- Programme detail -->
+                <section id="{{ key }}" class="programme-detail modal" tabindex="-1">
+                  <div class="programme-detail-inner">
+                    <div class="modal-actions">
+                      {% include '@modal-close' %}
+                    </div>
+
+                    <h2>{{ item.title }}</h2>
```
Comment (delrueba) [inline styleguide/components/41-organisms/programme/programme.twig]: Heading level is not correct
It places all programmes lots on the same level as the visually hidden programme title.

File: styleguide/components/41-organisms/programme/_programme.scss [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+      + .programme-detail-overlay {
+        display: block;
+      }
+    }
+
+    // Children
+    &-overlay {
+      @include theme('background-color', 'color-primary--darken-3', 'checkbox-with-filter-overlay-background-color');
+      display: none;
+
```
Comment (meirege) [inline styleguide/components/41-organisms/programme/_programme.scss]: Move this newline up a bit ;)

File: styleguide/components/41-organisms/programme/_programme.scss [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+        top: 1.2rem;
+        right: 1.2rem;
+      }
+    }
+
+    &-inner {
+      position: relative;
+      min-height: 100vh;
+      padding: 1rem;
+      background-color: color('white');
```
Comment (meirege) [inline styleguide/components/41-organisms/programme/_programme.scss]: Shouldn't this be themified? :)

File: styleguide/components/41-organisms/programme/_programme.scss [hunk 19]
```diff
@@ -0,0 +1,194 @@
+.programme {
+  width: calc(100% + 2 * #{$gutter-width});
+  margin-right: -$gutter-width;
+  margin-left: -$gutter-width;
+
+  > .inner-box {
```
Comment (delrueba) [inline styleguide/components/41-organisms/programme/_programme.scss:6]: Spot icon looks a bit funky
image

File: yarn.lock [hunk 20]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (meirege) [inline yarn.lock]: This file seems unnecessary?

File: styleguide/components/41-organisms/programme/_programme.scss [hunk 21]
```diff
@@ -0,0 +1,194 @@
...
+
+    & + & {
+      margin-top: 2.8rem;
+    }
+  }
+
+  ///
+  /// Programme teaser.
+  ///
+  &-teaser {
```
Comment (delrueba) [inline styleguide/components/41-organisms/programme/_programme.scss]: These don't work like they are supposed to (mainly links inside the teaser) markup is very different too.
Can you make a new teaser variant in 31/teasers and follow standard teaser markup?

File: styleguide/components/41-organisms/programme/programme.twig [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+                        'text': item.description
+                      } %}
+                   {% endif %}
+
+                    {% if item.video %}
+                      {% include '@video' with item.video %}
+                    {% endif %}
+                  </div>
+                </section>
+                <div class="programme-detail-overlay"></div>
```
Comment (delrueba) [inline styleguide/components/41-organisms/programme/programme.twig]: I believe you are missing the "modal__close" classe, because the modal doesn't close.

File: styleguide/components/41-organisms/programme/programme.twig [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+                <!-- /Programme detail -->
+
+                <!-- Programme Teaser -->
+                <article class="programme-teaser">
+                  <!-- Content -->
+                  <div class="programme-teaser__bottom">
+                    <h3 id="uid_{{ key }}">{{ item.title }}</h3>
+                    <span class="programme-teaser-subtitle">{{ item.subtitle }}</span>
+
+                    <a href="javascript:void(0);" class="read-more standalone-link" aria-labelledby="uid_{{ key }}">read more</a>
```
Comment (delrueba) [inline styleguide/components/41-organisms/programme/programme.twig]: Missing modal trigger class.
Unable to open the modal by keyboard.

**Review by delrueba — CHANGES_REQUESTED** [review]

**Comment by delrueba** [comment]
There are some extra overall issues
* modal is not consistent to for instance: checkbox-with-filter. Mainly because we never created a true modal component.... maybe it's time to do so ;)
** Missing shadow, modal-close isn't positioned right, overlay is not clickable, padding is a lot...

Add a window.onhashchange listener
* users will be able to navigate from one modal to another
* we don't have to bind a click event to a non functioning "read more" url, it'll actually be a url! How nice is that!

Why not make "changeHash" default true?
At least the history management, it will be confusing UX if one modal does this and the other does not.

File: styleguide/components/41-organisms/programme/_programme.scss (hunk 18 above)
Comment (delrueba) [reply styleguide/components/41-organisms/programme/_programme.scss]: Don't agree. White is white. This corresponds to the page background and that's also not themifiable.

File: styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss (hunk 6 above)
Comment (delrueba) [reply styleguide/components/41-organisms/image-collections/image-gallery/_image-gallery.scss]: I think that's legacy from me. I was a bit anal and liked the logic of 100% / 3, 100% / 2 ... 100% / 1.

File: styleguide/components/41-organisms/programme/_programme.scss (hunk 14 above)
Comment (delrueba) [reply styleguide/components/41-organisms/programme/_programme.scss:38]: I am not afraid to admit I had to look it up. :wink:

File: styleguide/components/41-organisms/image-collections/image-gallery/image-gallery--multiple.twig (hunk 9 above)
Comment (hstandaert) [reply styleguide/components/41-organisms/image-collections/image-gallery/image-gallery--multiple.twig:1]: Debatable... If we'd place the space before the if-statement, `<div class="image-gallery--wrapper multiple ">` will be rendered in the DOM for a image gallery with no type specified. And that's kind of weird with the trailing space, don't you think?

File: styleguide/components/41-organisms/programme/_programme.scss (hunk 12 above)
Comment (hstandaert) [reply styleguide/components/41-organisms/programme/_programme.scss:3]: Can't find one on first sight 🔍

File: styleguide/components/41-organisms/programme/_programme.scss (hunk 14 above)
Comment (hstandaert) [reply styleguide/components/41-organisms/programme/_programme.scss:38]: Don't know.. ? It's just the combination of: 

- The `+` selector in CSS, which is possibly not known by every developer but it should be. We're using the `>` selector as well?
- the `&`-sign in SASS, which is widely used throughout all SASS files in the repo.

I do acknowledge that it could be a weird construct if you see it for the first time, but I've seen it in numerous projects and use it a lot myself.

Wdyt? Keep it or throw it out?

Comment (delrueba) [reply styleguide/components/41-organisms/programme/_programme.scss:38]: Keep it, no need to comment for me either.

File: theme.inc [hunk 24]
```diff
@@ -8,6 +8,7 @@
 use Drupal\file\Entity\File;
 use Drupal\media\Entity\Media;
 use Drupal\node\Entity\Node;
+use Drupal\paragraphs\Entity\Paragraph;
 
 /**
  * Check to see if a give view mode exists.
```
Comment (MPParsley) [inline theme.inc]:
```suggestion
 * Check to see if a given view mode exists.
```

File: theme.inc [hunk 25]
```diff
@@ -69,7 +70,10 @@ function _gent_base_get_entity_reference_file_data(&$variables, $field) {
   }
 
   $file_name = $file_entity->getFilename();
-  $file_type = strtoupper(end(explode('.', $file_name)));
+
+  $file_type = explode('.', $file_name);
+  $file_type = end($file_type);
+  $file_type = strtoupper($file_type);
```
Comment (MPParsley) [inline theme.inc:75]: Moeten we dit in PHP doen of kan dit in CSS?

Comment (hstandaert) [reply theme.inc:75]: @MPParsley, zoals besproken: we nemen dit op in de algemene refactoring van de `gent_base.theme` file 😉

Comment (MPParsley) [reply theme.inc:75]: :ok_hand:

**Review by delrueba — APPROVED** [review]
